### ISSUE TITLE
layout_with_kk produces NaNs

### ISSUE THREAD

**Issue by rcannood** [igraph/igraph#1928]
A very specific numerical value generates NaNs in layout_with_kk:
```R
library(igraph)
gr <- graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043116))
layout_with_kk(gr)
```

```
     [,1]         [,2]
[1,]  NaN          NaN
[2,]   -1 1.224647e-16
```

Numbers in the same range do not result in NaNs being produced:
```
> layout_with_kk(graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043115)))
           [,1]         [,2]
[1,]  0.4142136 3.586907e-17
[2,] -1.0000000 1.224647e-16
> layout_with_kk(graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043116)))
     [,1]         [,2]
[1,]  NaN          NaN
[2,]   -1 1.224647e-16
> layout_with_kk(graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043117)))
          [,1]          [,2]
[1,] -0.632981 -1.365759e+00
[2,] -1.000000  1.224647e-16
```

Multiplying by a power of 2 also results in NaNs, until a certain point:
```
> layout_with_kk(graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043116 * 2)))
     [,1]         [,2]
[1,]  NaN          NaN
[2,]   -1 1.224647e-16
> layout_with_kk(graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043116 * 2^10)))
     [,1]         [,2]
[1,]  NaN          NaN
[2,]   -1 1.224647e-16
> layout_with_kk(graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043116 * 2^-3)))
     [,1]         [,2]
[1,]  NaN          NaN
[2,]   -1 1.224647e-16
> layout_with_kk(graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043116 * 2^-4)))
     [,1]         [,2]
[1,]  NaN          NaN
[2,]   -1 1.224647e-16
> layout_with_kk(graph_from_data_frame(data.frame(from = "a", to = "b", weight = 5.272340994043116 * 2^-5)))
           [,1]          [,2]
[1,] -0.6000947 -1.356494e+00
[2,] -1.0000000  1.224647e-16
```

What could be the problem here?

cc: @zouter

**Comment by szhorvat** [comment]
This is reproducible with the Mathematica interface, moving the the C core repo.

**Comment by rcannood** [comment]
Afaik this issue still needs to be fixed.

**Comment by szhorvat** [comment]
I suspect that with some specific edge weight values, two vertices will be moved almost precisely on top of each other, which then causes instability. I have not tested this hypothesis.

**Comment by szhorvat** [comment]
How did you come up with this very specific weight value, @rcannood ? Does this problem appear in any practical example, or is it artificially generated?

**Comment by rcannood** [comment]
This weight value was defined from the analysis in which I encountered the issue. I don't recall what I was doing at the time, but I trimmed the input data to the minimal reproducible example before posting the issue.

**Comment by szhorvat** [comment]
I made a change to the KK layout to fix #963. As a side effect, your original example no longer produced NaNs. This does not mean that the problem is fixed. It is possible that there exist other weights which trigger the problem.

**Comment by szhorvat** [comment]
To be specific, I expect that there exists a weight for a two-vertex graph which will result in the two vertices overlapping after the first move.  That could cause NaN/Infs. But this should happen only for a single, very specific weight value. It is extremely unlikely that someone would hit that specific value by accident. Thus I am not sure that the issue reported here was due to the same situation.

**Comment by szhorvat** [comment]
This is fixed now, see #2074.
